### DISCUSSION TITLE
Status of this crate

### DISCUSSION THREAD

**Discussion by itamarst** [anweiss/cddl#113] [Q&A] [answered]
Hi,

Thanks for publishing this crate!

I'm investigating CDDL validation for Python. Lacking an existing library in Python, wrapping a safe implementation in Rust seems like a much better approach than C/C++,  and your library came up. The README has a caveat ("personal learning exercise" etc.), but:

1. You still seem to be actively working on this, and sounds like trying to get to 1.0?
2. The other Rust alternative is `cddl-cat` (https://github.com/ericseppanen/cddl-cat) which seems a lot less featureful.
3. There doesn't seem to be `unsafe` so presumably there's a limit to how much can go wrong (a panic, worst case? and I believe the Python wrapper will just turn that into a Python exception).

So perhaps that warning is no longer valid? In which case, perhaps it should be removed.

**Comment by anweiss** [comment, marked as answer]
Great question @itamarst! My original intent for the warning was more of a "catch-all"  to point folks to potential alternative libraries when I began working on this a few years ago. However, at this point, and as you mentioned, I am indeed trying to get this to a stable 1.0 release and plan to remove the warning soon.

While https://github.com/ericseppanen/cddl-cat may not have as many features implemented, at the moment I think its source code is a bit more readable than my own lol. Although this is something I continue to clean up over time. That crate also uses the nom parser generator vs. a handcrafted parser, which may/may not have advantages. Tracking this in https://github.com/anweiss/cddl/issues/16.

And indeed, there is no unsafe code in this crate.

Of course, any and all contributions are welcome!

**Comment by itamarst** [comment]
Thank you!
